content/ai/concepts/prompt-engineering/prompt-engineering.md

```diff
@@ -0,0 +1,99 @@
+  - 'intro-to-chatgpt'
+    # @KTom101, please let me know if this would be the right path
+  - 'paths/intro-to-chatgpt'
+---
+
+**Prompt Enginneering** is a low-code solution to providing instructions to Artifical Intelligences(AIs). AIs are typically trained with large amounts of data. As a result, when an AI is given a task, it uses all this data to complete the task. Keep in mind, the task probably does not require that much data. This introduces a handful of issues. Such as hallucinations, or possibly using data that is irrelevant to completing the task. Prompt Engineering allows users to clearly define the guard rails in which the AI must follow in completing the task assigned to it.
+
+## General Tips to Writting a Good Prompt
+
+### Use Clear Language
+
+When forming a prompt, it is important to be precise, yet concise in your instruction. This means being specific, while also trying to do so in the least amount of words. There are several benefits when following this strategy. The motivation for this requires a minimal understanding of how Large Language Models (LLMs).
+
+When an LLM is given a body of text it typically breaks this text into individual units or "tokens." It then processes these tokens in order to generate an answer. This computation comes at a cost, and is the reason why companies charge a certain amount of dollars per token. Particularly when these tokens are being sent through some sort of API request. In these situations, having clear language can not only save money, but also directly affect the performance of the LLM.
+
+One example of using clear language could be the following.
+
+```
+Explain the concept of for loops in Javascript. Keep your explaination short, and do not be too descriptive.
+```
+
+In this example, there is too much ambiguity for an LLM. For example, there is no guidance on what "short" means, and "not be too descriptive" could have two meanings. Either, do not use too many words, which we already said, or provide a description that is easy to understand. A better prompt would be the following.
+
+```
+Use 2 to 3 sentences to explain the concept of for loops in Javascript to a 5 year old.
+```
+
+### Define Your Purpose
+
+Another key to writting effective prompts, is defining the purpose of your prompt. Or in other words, effectively defining the goal you wish for the LLM to achieve. This can be done by specifying the tone, format and audience of your prompt.
+
+One area in which tone is important, is writting an email. For example, you would probably have a different tone when writting an email to your sibling, versus your boss. Therefore, when asking an LLM to write a draft of an email, it is important to make sure it understands what tone you would prefer.
+
+In addition to specifying the tone, you can also specify the format that you wish an LLM to follow. In some situations you may want the LLM to use a bullet list, versus a series of paragraphs.
```

Comment:
```suggestion
In addition to specifying the tone, the format that an LLM will follow can be specified. In some situations it can be specified that the LLM use a bullet list, versus a series of paragraphs.
```